tool/extract_deps.py

```diff
@@ -400,25 +398,29 @@ def run_commands_in_sequence(commands, initial_input=None):
                 "-Dverbose",
                 "-DoutputType=json",
                 "-DoutputFile=/dev/stdout",
-                "-f",
-                temp_pom_path,
             ],
-            ["grep", "-v", "\\[INFO"],
-            ["jq", "[.children | .. | {version, groupId, artifactId}?] | unique"],
+            ["grep", "-v", "^\\[\\(INFO\\|WARNING\\)"],
+            # Below: In some cases, for parent packages which need to access child pom files, the above returns several JSON objects
+            # This requires it to be "slurped" into a single JSON array, as the next command expects a single JSON object
+            ["jq", "--slurp", "[.[] | .children[]?] | unique | map({version, groupId, artifactId})"],
```

Comment:
I believe you could output `dependency:tree` (`-DoutputFile=/tmp/deps.json`) to a file and then parse JSON. RegEx is too error-prone.

Reply:
Also, could you or @LogFlames confirm that output of `mvn dependency:tree` is independent of passing status of build? If it is not, you may get a partial tree if the build fails.

Reply:
@algomaster99 I was looking at `dependency`'s goals, and from those we want to use here, 'tree' is the only one with JSON outputs, I believe. As such, I was thinking of _not_ outputting JSON, for consistency and making it the same function to handle both calls -- what's your opinion?
At the same time though, the current function is definitely over-engineered, so I'm going to simplify it, and ping you in a new PR when I open it.

Regarding the build status question, I am not sure, but I'm going to investigate